### PR TITLE
Add curp steps

### PR DESCRIPTION
**Pull request by rogercloud**
The recovery part is not complete

### CODE REVIEW

**Review by pzheng1025 — APPROVED** [review]

File: curp/doc/steps.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+1. Check `TermId` 
+    - if `TermId` doesn't match, reply tuple <WrongTermId> back to the client. 
+    - otherwise continue.
+2. Check `WitnessListVersion`
+    - if `WitnessListVersion` doesn't match, reply tuple <WrongWitnessListVersion, WitnessListVersion>
+    - otherwise continue.
+3. Check whether `cmd` conflicts with any cmd in the witness
+    - if there's no confliction, put the cmd in the witness, record tuple <Cmd, RpcId>, reply tuple <Acceptted> 
+    - otherwise reply typle <NeedSync>
```
Comment (pzheng1025) [inline curp/doc/steps.md:51]: tuple

File: curp/doc/steps.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
+The document describes the CURP steps in detail.
+
+Definition:
+1. The request tuple consists of request command and following parameters.
+2. The response tuple consists of return code and following states.
+
+# Normal Operation
+## A. Client Send Propose
+
+1. Client send request <Propose, Cmd, RpcId, TermId, WitnessListVersion> to the master and all witnesses.
```
Comment (pzheng1025) [inline curp/doc/steps.md:10]: What is the difference between Propose and Cmd?

File: curp/doc/steps.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+## B. Master Handle Client Propose
+
+Request: <Propose, Cmd, RpcId, WitnessListVersion>
+
+1. Check `TermId`
+    - if `TermId` doesn't match, reply tuple <WrongTermId> back to the client. 
+    - otherwise continue.
+2. Check `WitnessListVersion`
+    - if `WitnessListVersion` doesn't match, reply tuple <WrongWitnessListVersion, WitnessListVersion>
```
Comment (pzheng1025) [inline curp/doc/steps.md:26]: also reply with the current witness list?

Comment (rogercloud) [reply curp/doc/steps.md:26]: Witness list comes from the configuration server, so we don't have to reply the witness list itself.

File: curp/doc/steps.md (hunk 2 above)
Comment (rogercloud) [reply curp/doc/steps.md:10]: Propose is the RPC type, Cmd is the request itself.

**Pull request merged**